### ISSUE TITLE
String prepend methods

### ISSUE THREAD

**Issue by elland** [bakkenbaeck/SweetSwift#32]
Appending is all nice and well, but what about prepending? 

Would something like that interest you?

``` swift
extension String {
    mutating func prepend(_ aString: String) {
        self = aString.appending(self)
    }

    func prepending(_ aString: String) -> String {
        return aString.appending(self)
    }
}
```

**Comment by designatednerd** [comment]
Yeah that seems reasonable. Although is there a mutating `append` that would make more sense to use than `self = aString.appending(self)`?

**Comment by elland** [comment]
@designatednerd what would that look like?

**Comment by designatednerd** [comment]
Ah OK, trying to come up with an example I see why that wouldn't work - you'd still have to assign `aString` to self since you'd be mutating `aString` if you just used `aString.append`
